### ISSUE TITLE
Error occured when opening file (Fedora / Linux)

### ISSUE THREAD

**Issue by 13k** [careteditor/caret#583]
It happened after these steps:

1. Open Caret, most recent file automatically opened
2. List the files in the current file's directory (Ctrl-T)
3. Canceled the `Files` list
4. List recent files (Ctrl-E)
5. Open one of the files in the `Recents` list

After the error happened, I kept trying to reproduce and noticed the `Files` (Ctrl-T) listing (from step 2) was still running and actually indexing my HOME directory (file opened in step 1 was in my HOME directory) and was still updating the list.

While the indexing was going, the `Files` list was inaccessible (couldn't select any files) and editing files was hampered.

After the indexing was finished, the editor returned to regular operation with no errors.

It's worth mentioning that the `Files` listing was matching every file in the directory traversal (gzips, pdfs, etc) and looks like at some point it was cut off (presumably because of the error). Subsequent `Files` activations always returned the short list (probably cached).

--- 

The following error has occurred:
```
{
    "message": "Cannot read property 'find' of undefined",
    "version": "3.4.6",
    "os": "Linux 4.13",
    "path": "extensions/Project.js",
    "code": "n=r.items.find(function(t)",
    "line": 5,
    "column": 3102,
    "stack": [
        "at n (extensions/Project.js:5:3102)",
        "at t.value (extensions/Project.js:5:3261)",
        "at t.value (extensions/Project.js:5:2674)",
        "at t.value (extensions/Navigator.js:5:2196)",
        "at e.value (web/List.Filter.js:5:1823)",
        "at /opt/caret/resources/app.asar/extensions/Navigator.js:5:1943"
    ]
}
```

**Comment by astoilkov** [comment]
Hi @13k,

We will investigate this problem. Thanks for reporting.

**Comment by astoilkov** [comment]
This issue is now fixed in the latest version of Caret 4.0 Beta available for download from [our site](https://caret.io).

Closing this issue. Thanks for the detailed description!
